test/test_pascal.c

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2022 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include "demangling_unit.h"
+
+mu_demangle(0, pascal, "OUTPUT_$$_SQUARE$SMALLINT$$SMALLINT", "unit output square(var a0: smallint) smallint");
+mu_demangle(1, pascal, "OUTPUT_$$_init", "unit output init()");
+mu_demangle(2, pascal, "OUTPUT$_$MYOBJECT_$__$$_INIT$$QWORDBOOL", "unit output myobject.init() qwordbool");
+mu_demangle(3, pascal, "OUTPUT$_$MYOBJECT_$__$$_MYMETHOD", "unit output myobject.mymethod()");
+mu_demangle(4, pascal, "OUTPUT_$$_MYFUNC$$POINTER", "unit output myfunc() pointer");
+mu_demangle(5, pascal, "OUTPUT_$$_MYPROCEDURE$SMALLINT$LONGINT$PCHAR", "unit output myprocedure(var a0: smallint; a1: longint; a2: pchar)");
+mu_demangle(6, pascal, "OUTPUT_$$_MYFUNC2$SMALLINT$LONGINT$PCHAR$$POINTER", "unit output myfunc2(var a0: smallint; a1: longint; a2: pchar) pointer");
```

Comment:
kinda unsure if is better to simplify this to `myfunc2(smallint,longint,pchar)` like fpc does.

Reply:
Both variants are fine, I think.